baseapp/baseapp.go

```diff
@@ -510,6 +513,19 @@ func handleQueryCustom(app *BaseApp, path []string, req abci.RequestQuery) (res
 	}
 }
 
+func (app *BaseApp) validateHeight(req abci.RequestBeginBlock) error {
+	if req.Header.Height < 0 {
+		return fmt.Errorf("invalid height: %d", req.Header.Height)
+	}
+
+	prevHeight := app.LastBlockHeight()
+	if prevHeight != 0 && req.Header.Height != prevHeight+1 {
```

Comment:
Why do we need to check `prevHeight != 0`?

Reply:
If the current height is 0, then presumably, the prevHeight has not yet been initialized and is 0?

Reply:
Ahh, it was because certain unit tests were failing, but that was because the proper header was not being used. Updated.